### DISCUSSION TITLE
Codespace doesn't handle python venv properly

### DISCUSSION THREAD

**Discussion by kmahyyg** [community/community#25452] [Codespaces]
<p>Your code just added a folder called <code>pythonenv3.8</code> under my project folder… Since the most common and default venv folder name should be <code>.venv</code> or <code>venv</code>, it will NOT be ignored by git…</p>
<p>Please change this behaviour…</p>

**Comment by jkeech** [comment, marked as answer]
<p><a class="mention" href="/kmahyyg">@kmahyyg</a>, thanks for the feedback! This is coming from Oryx here: <a href="https://github.com/microsoft/Oryx/blob/1a35fbce482b20b71290f3a837a3469803ce4b44/src/BuildScriptGenerator/Python/PythonPlatform.cs#L466" rel="noopener nofollow ugc">https://github.com/microsoft/Oryx/blob/1a35fbce482b20b71290f3a837a3469803ce4b44/src/BuildScriptGenerator/Python/PythonPlatform.cs#L466</a>. We run Oryx automatically if you do not specify a <code>postCreateCommand</code> in <code>devcontainer.json</code>. Oryx inspects your repo, infers the project type, and then restores and builds the project if it’s one of the languages/frameworks that it understands.</p>
<p>One way to workaround this is to define a <code>postCreateCommand</code> in your repo which could do any sort of setup you’d like after the container is created instead of falling back to the default inference behavior.</p>
